src/pages/storage/actions/DeleteStorageVolumeBtn.tsx

```diff
@@ -104,8 +106,12 @@ const DeleteStorageVolumeBtn: FC<Props> = ({
       className={classname}
       shiftClickEnabled
       showShiftClickHint
-      disabled={Boolean(disabledReason)}
-      onHoverText={disabledReason}
+      disabled={!canDeleteVolume(volume) || Boolean(disabledReason)}
+      onHoverText={
+        canDeleteVolume(volume)
+          ? ""
+          : "You do not have permission to delete storage volumes."
```

Comment:
```suggestion
          : "You do not have permission to delete this volume."
```
Better be precise, permissions can be granted on a entity level.